API/FrontEnd/templates/includes/cards.html

```diff
@@ -0,0 +1,77 @@
+{% macro cards_w_img(products) %}
+<div class="card-deck row-cols-md-3">
+    
+    {% for product in products %}
+        
+    <div class="mb-3">
+        <div class="card text-center"  style="background-color: #10D48E;">
+            <img src= "{{product.url}}" class="card-img-top" height=400 width=400 alt="...">
+            <div class="card-body">

```

Comment:
```suggestion
            <div class="card-body" style=" white-space: nowrap;overflow: hidden; text-overflow: ellipsis;">

```

Reply:
There is an incompatibility between the cards due to their card-title size.
You can add text-overflow property to card-title in order to eliminate the size differences.
This provides card-title to have only one text line.